.github/workflows/publish.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  push:
+    branches:
+      - main
+      - ci
+  schedule:
+    - cron: 0 0 * * *
+name: Build & Publish
+jobs:
+  build:
+    env:
+      GLOBAL_GEMS: 1
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+          cache-version: 2
```

Comment:
I think this will need to specify the version explicitly to stay on the right version (2.7) for now?

Reply:
No, it reads from `.ruby-version`